### ISSUE TITLE
Training faces descriptors of one person vs query face descriptor

### ISSUE THREAD

**Issue by SaifEddineGhribi** [justadudewhohacks/face-api.js#657]
Hello, 

I'm wondering  for one person with many training descriptors(training images) ,when we want to recognize a new query face image ,will the recognize method take the average of the training descriptors of every single person and compare it with the query face descriptor ? 

Thanks in advance.

**Comment by karthik-Gopalan** [comment]
I don't believe it does a 'weighted average'. I believe it goes through each face given in the descriptors. You add more faces to ensure that you have better references for each person. What is your specific use case, if you can elaborate the same?

**Comment by SaifEddineGhribi** [comment]
> I don't believe it does a 'weighted average'. I believe it goes through each face given in the descriptors. You add more faces to ensure that you have better references for each person. What is your specific use case, if you can elaborate the same?

I'm trying to understand how the model works to ensure I have the best performance.So I want to know if adding more faces to a person increases the recognition performance of that person or the model just compares the query face descriptor with all the face descriptors of that person (which means a bad picture might mislead the model as well ) ?

**Comment by karthik-Gopalan** [comment]
When you add a reference image to the descriptor, and run an image through the api, the api does not know which person it is, and hence it will run through all the persons defined in the descriptor. Is parallel processing possible? If you have thousands of reference images, it might make sense to parallel process the same. I don't know, as I have not tried it.
